dataprofiler/data_readers/graph_data.py

```diff
@@ -0,0 +1,159 @@
+            
+            # fetch only column names
+            for row in csv_reader:
+                column_names.append(row)
+                break
+        column_names = column_names[0]
+
+        # replace all whitespaces in the column names
+        for index in range(0, len(column_names)):
+            column_names[index] = column_names[index].replace(" ", "")
+
+        return column_names
+
+
+    @classmethod
+    def is_match(cls, file_path, options=None):
+        '''
+        Determines whether the file is a graph
+        Current formats checked:
+            - attributed edge list
+
+        This works by finding whether the file contains a target and a source node
+        '''
+        if options is None:
+            options = dict()
+        if not CSVData.is_match(file_path, options):
+            return False
+        column_names = cls.csv_column_names(file_path, options)
+        source_keywords = ['source', 'src', 'origin']
+        target_keywords = ['target', 'destination', 'dst']
+        source_index = cls._find_target_string_in_column(column_names, source_keywords)
+        destination_index = cls._find_target_string_in_column(column_names, target_keywords)
+        has_source = True if source_index >= 0 else False
+        has_target = True if destination_index >= 0 else False
+
+        if has_target and has_source:
+            options.update(source_node = source_index)
+            options.update(destination_node = destination_index)
+            options.update(destination_list = target_keywords)
+            options.update(source_list = source_keywords)
+            options.update(column_name = column_names)
+            return True
+
+        return False
+
+    def _format_data_networkx(self):
+        '''
+        Formats the input file into a networkX graph
+        '''
+        source_keywords = ['source', 'src', 'origin']
+        target_keywords = ['target', 'destination', 'dst']
+
+        if self.input_file_path:
+            column_names = self.csv_column_names(self.input_file_path, self.options)
+
+            if self.options.get("source_node") is None or self.options.get("destination_node") is None:
+                source_node = self._find_target_string_in_column(column_names, source_keywords)
+                destination_node = self._find_target_string_in_column(column_names, target_keywords)
+            else:
+                source_node = self.options.get("source_node")
+                destination_node = self.options.get("destination_node")
+
+            # read lines from csv
+            csv_as_list = []
+            with FileOrBufferHandler(self.input_file_path) as csv_file:
+                csv_reader = csv.reader(csv_file, delimiter = self.options.get("delimiter", self.options.get("delimiter", ",")))
+                for row in csv_reader:
+                    row = [item.replace(" ", "") for item in row]
+                    csv_as_list.append(row)
```

Comment:
maybe we should just do the same as CSVData here.
```python
data_utils.read_csv_df(
    input_file_path,
    self._delimiter, self._header, self._column_names,
    read_in_string=True,
    encoding=self.file_encoding
)
```

Reply:
then we can load from pandas_edgelist? or loop through as you desire

Reply:
https://networkx.org/documentation/stable/reference/generated/networkx.convert_matrix.from_pandas_edgelist.html#networkx.convert_matrix.from_pandas_edgelist
Not sure what versions it is viable within